dwave/cloud/utils.py

```diff
@@ -801,3 +802,197 @@ def bqm_to_dqm(bqm):
+    depends on a QPU’s working graph, such embeddings should be for
+    the particular QPU for which you are estimating the access time.
+    See examples below.
+
+    Args:
+        num_qubits:
+            Number of qubits required to represent your binary quadratic model
+            on the selected solver.
+        num_reads:
+            Number of reads. Provide this value if you explictly set ``num_reads``
+            in your submission.
+        annealing_time:
+            Annealing duration. Provide this value of if you set
+            ``annealing_time`` in your submission.
+        anneal_schedule:
+            Anneal schedule. Provide the ``anneal_schedule`` if you set it in
+            your submission.
+        initial_state:
+            Initial state. Provide the ``initial_state`` if your submission
+            uses reverse annealing.
+        reinitialize_state:
+            Set to ``True`` if your submission sets ``reinitialize_state``.
+        programming_thermalization:
+            programming thermalization time. Provide this value if you explictly
+            set a value for ``programming_thermalization`` in your submission.
+        readout_thermalization:
+            Set to ``True`` if your submission sets ``readout_thermalization``.
+        reduce_intersample_correlation:
+            Set to ``True`` if your submission sets ``reduce_intersample_correlation``.
+
+    Returns:
+        Estimated QPU access time, in microseconds.
+
+    Raises:
+        KeyError: If a solver property, or a field in the ``problem_timing_data``
+            solver property, required by the timing model is missing for the
+            selected solver.
+
+        ValueError: If conflicting parameters are set or the selected solver
+            uses an unsupported timing model.
+
+    Examples:
+
+        This example estimates the QPU access time for a selected quantum computer
+        for a problem that uses 500 qubits and is sampled with 100 anneals.
+
+        >>> from dwave.cloud.utils import estimate_qpu_access_time
+        >>> from dwave.cloud import Client
+        ...
+        >>> with Client.from_config() as client:
+        ...     solver = client.get_solver()
+        ...     estimated_runtime = estimate_qpu_access_time(solver, 500, num_reads=100) # doctest: +SKIP
+        ...     print("%.0f" % estimated_runtime) # doctest: +SKIP
+        42657
+    """
+    if anneal_schedule and annealing_time:
+        raise ValueError("set only one of ``anneal_schedule`` or ``annealing_time``")
+
+    if anneal_schedule and anneal_schedule[0][1] == 1 and not initial_state:
+        raise ValueError("reverse annealing requested (``anneal_schedule`` "
+                         "starts with s==1) but ``initial_state`` not provided")
+
+    try:
+        problem_timing_data = solver.properties['problem_timing_data']
+
+        version_timing_model = problem_timing_data['version']
```

Comment:
I suggest a slight re-org here:
- split this big try/catch in three
- in first, test for `problem_timing_data` (or fail appropriately) -> no ugly switch in except block
- in second, test version (or fail appropriately)
- in third, read out the required properties (or fail)